astropy/io/fits/tests/test_checksum.py

```diff
@@ -83,11 +83,52 @@ def test_scaled_data(self):
                 assert 'DATASUM' in hdul1[0].header
                 assert hdul1[0].header['DATASUM'] == '1891563534'
 
+    def test_scaled_data_auto_rescale(self):
+        """
+        Regression test for
+        https://github.com/astropy/astropy/issues/3883#issuecomment-115122647
+
+        Ensure that when scaled data is automatically rescaled on
+        opening/writing a file that the checksum and datasum are computed for
+        the rescaled array.
+        """
+
+        with fits.open(self.data('scale.fits')) as hdul:
+            # Write out a copy of the data with the rescaling applied
+            hdul.writeto(self.temp('rescaled.fits'))
+
+        # Reopen the new file and save it back again with a checksum
+        with fits.open(self.temp('rescaled.fits')) as hdul:
+            hdul.writeto(self.temp('rescaled.fits'), clobber=True,
+                         checksum=True)
```

Comment:
Interesting--this part right here seems to have revealed an unrelated bug that on Windows a file can't be opened, and then overwritten to the same file while open.  I'll probably just change the test for now, but open a separate issue for this.